### PR TITLE
Feature/cxp-873 TypeScript convert table component

### PR DESCRIPTION
**Pull request by yasskin**
## PR Checklist

Storybook can be viewed [here](https://docspot.adnxs.net/projects/lucid/feature_CXP-873-ts-convert-table/?path=/docs/table-table--standard)

- Manually tested across supported browsers

  - [x] Chrome
  - [x] Firefox
  - [x] Safari

- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- [x] One core team UX approval

### CODE REVIEW

File: src/components/ScrollTable/ScrollTable copy.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import PropTypes from 'react-peek/prop-types';
+import { createClass, omitProps } from '../../util/component-types';
+import { lucidClassNames } from '../../util/style-helpers';
+import Table from '../Table/Table';
+
+const cx = lucidClassNames.bind('&-ScrollTable');
+
+const { object, string, bool, node, number, oneOfType } = PropTypes;
+
+const ScrollTable = createClass({
```
Comment (jhsu) [inline src/components/ScrollTable/ScrollTable copy.jsx]: did you mean to include this file? `ScrollTable copy.jsx`

File: src/components/Table/Table_copy.jsx [hunk 2]
```diff
@@ -25,27 +25,6 @@ const {
 } = PropTypes;
 
 const Thead = createClass({
-	displayName: 'Table.Thead',
```
Comment (jhsu) [inline src/components/Table/Table_copy.jsx]: did you mean to include this file `Table_copy.jsx`?

File: src/components/ScrollTable/ScrollTable copy.jsx (hunk 1 above)
Comment (yasskin) [reply src/components/ScrollTable/ScrollTable copy.jsx]: Good eye. Nope. That will come out.

File: src/components/Table/Table_copy.jsx (hunk 2 above)
Comment (yasskin) [reply src/components/Table/Table_copy.jsx]: Ugh. Deleted that now too. Thanks, Joe!

File: src/components/ScrollTable/ScrollTable copy.jsx (hunk 1 above)
Comment (yasskin) [reply src/components/ScrollTable/ScrollTable copy.jsx]: Nope. Good catch.

File: src/components/Table/Table_copy.jsx (hunk 2 above)
Comment (yasskin) [reply src/components/Table/Table_copy.jsx]: Nope. Took it out. Thx.

**Comment by rsadres** [comment]
Hey!
Looks good!
Here are some things I noticed:
- Missing the blue vertical bar on row hover
- row height seems taller than the existing version - not sure why
- compressed example also seems to have a larger row height than the existing compressed example
- empty column example seems odd - I thought we used to use an image in the no data cells - not the words no data. what happened to that?
- the no hover example is off - even in the existing version - it still shows a hover state for the selected rows.

I was comparing it to the existing one - https://appnexus.github.io/lucid/?path=/story/table-table--standard

File: src/components/Table/Table.less [hunk 3]
```diff
@@ -256,39 +267,52 @@
...
 			// style row cells on hover
 			&:hover {
-				.selectCellsOnRow({ background-color: @color-table-row-hover-background;});
-			};
+				.selectCellsOnRow(
+					{background-color: @color-table-row-hover-background;}
+				);
+			}
+
+			// &:hover > .@{prefix}-Table-Td {
```
Comment (dreidn) [inline src/components/Table/Table.less]: Can we delete this comment?

File: src/components/Table/Table.tsx [hunk 4]
```diff
@@ -0,0 +1,1100 @@
...
+	return finalGrid;
+}
+
+/**
+ * renderRowsWithIdentifiedEdges
+ *
+ * Returns an equivalent list of Tr's where each cell on the perimeter has props set for: `isFirstRow`, `isLastRow`, `isFirstCol`, `isLastCol`, and `isFirstSingle`
+ */
+
+//{renderRowsWithIdentifiedEdges(filterTypes(children, Tr), Th)}
```
Comment (dreidn) [inline src/components/Table/Table.tsx]: might be able to remove this comment too

File: src/components/Table/Table.less (hunk 3 above)
Comment (yasskin) [reply src/components/Table/Table.less]: done

File: src/components/Table/Table.tsx (hunk 4 above)
Comment (yasskin) [reply src/components/Table/Table.tsx]: good catch. done.

**Pull request merged**